src/components/UI/SelectTokenRow/SelectTokenRow.js

```diff
@@ -23,17 +23,19 @@ export const SelectTokenRow = ({tokenData, onClick}) => {
             <div className={styles.name}>{name}</div>
           </div>
         </div>
-        <div className={styles.right}>
-          <>
-            {!isLoading ? (
-              <div className={styles.balance}>
-                {formatBalance(balance)} {symbol}
-              </div>
-            ) : (
-              <Loading size={LoadingSize.SMALL} />
-            )}
-          </>
-        </div>
+        {showBalance && (
+          <div className={styles.right}>
+            <>
+              {!isLoading ? (
+                <div className={styles.balance}>
+                  {formatBalance(balance)} {symbol}
+                </div>
+              ) : (
+                <Loading size={LoadingSize.SMALL} />
+              )}
+            </>
+          </div>
+        )}
```

Comment:
I'm not sure we need a new prop for this.
We can decide based on the `balance` value.
If the `balance === undefined` then the user isn't login (except the case this is the first balance fetching but then `isLoading=true`).

Reply:
fixed